master/internal/api_trials.go

```diff
@@ -947,9 +947,13 @@ func (a *apiServer) AllocationPreemptionSignal(
 
 	id := uuid.New()
 	var w task.PreemptionWatcher
-	if err := a.ask(handler.Address(), task.WatchPreemption{
+	err = a.ask(handler.Address(), task.WatchPreemption{
 		ID: id, AllocationID: allocationID,
-	}, &w); err != nil {
+	}, &w)
+
+	if _, ok := err.(task.ErrAllocationUnfulfilled); ok {
```

Comment:
i think we may need this logic in a few more places, too (anywhere that calls GetAllocationHandler).